source/mod_particles.f90

```diff
@@ -20,55 +18,133 @@ end subroutine part_allocate
 subroutine part_expand
 end subroutine part_expand
 
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Last modified: 2018-10-31
+!  Moved from maincr. Applies the closed orbit correction to the particle coordinates.
+! ================================================================================================ !
+subroutine part_applyClosedOrbit
+
+  use mod_common
+  use mod_commons
+  use mod_commonmn
+
+  implicit none
+
+  integer j
+
+  if(iclo6 == 2) then
+    xv1(1:napx)   = xv1(1:napx)   +  clo6v(1,1:napx)
+    yv1(1:napx)   = yv1(1:napx)   + clop6v(1,1:napx)
+    xv2(1:napx)   = xv2(1:napx)   +  clo6v(2,1:napx)
+    yv2(1:napx)   = yv2(1:napx)   + clop6v(2,1:napx)
+    sigmv(1:napx) = sigmv(1:napx) +  clo6v(3,1:napx)
+    dpsv(1:napx)  = dpsv(1:napx)  + clop6v(3,1:napx)
+  else if(idfor == 0) then
+    xv1(1:napx)   = xv1(1:napx)   +   clov(1,1:napx) * real(idz(1),fPrec)
+    yv1(1:napx)   = yv1(1:napx)   +  clopv(1,1:napx) * real(idz(1),fPrec)
+    xv2(1:napx)   = xv2(1:napx)   +   clov(2,1:napx) * real(idz(2),fPrec)
+    yv2(1:napx)   = yv2(1:napx)   +  clopv(2,1:napx) * real(idz(2),fPrec)
+  end if
+  call part_updatePartEnergy(3)
+
+end subroutine part_applyClosedOrbit
+
 ! ================================================================================================ !
 !  K.N. Sjobak, V.K. Berglyd Olsen, BE-ABP-HSS
-!  Last modified: 2018-08-13
-!  Updates the relevant particle arrays after the particle energies or the reference energy changed.
-!  If only the ejv array has been changed, pass e0 to refEnergy
+!  Last modified: 2018-11-02
+!  Updates the reference energy and momentum
 ! ================================================================================================ !
-subroutine part_updateEnergy(refEnergy)
+subroutine part_updateRefEnergy(refEnergy)
 
   use mod_hions
   use mod_common
-  use mod_commont
   use mod_commonmn
+  use numerical_constants
 
   implicit none
 
   real(kind=fPrec), intent(in) :: refEnergy
 
   real(kind=fPrec) e0o, e0fo
-  integer          j
+
+  if(e0 == refEnergy) return
+
+  ! Save previous values
+  e0o    = e0
+  e0fo   = e0f
 
   ! Modify the reference particle
-  e0o  = e0
-  e0fo = e0f
-  if(e0 /= refEnergy) then
-    e0     = refEnergy
-    e0f    = sqrt(e0**2 - nucm0**2)
-    gammar = nucm0/e0
+  e0     = refEnergy
+  e0f    = sqrt(e0**2 - nucm0**2)
+  gammar = nucm0/e0
+
+  ! Also update sigmv with the new beta0 = e0f/e0
+  sigmv(1:napx) = ((e0f*e0o)/(e0fo*e0))*sigmv(1:napx)
+
```

Comment:
I think we should scale also `rvv` the same way

Reply:
Can try, but I wanted to avoid using code that changed canonicals, so I kept the previous calculations. Not sure whether this will do that.

Reply:
I don't have a clear overview on `rvv`... @rdemaria ?